src/core/Table/TablePaginator.tsx

```diff
@@ -0,0 +1,335 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import React from 'react';
+import cx from 'classnames';
+import '@itwin/itwinui-css/css/table.css';
+import { IconButton } from '../Buttons/IconButton';
+import SvgChevronLeft from '@itwin/itwinui-icons-react/cjs/icons/ChevronLeft';
+import SvgChevronRight from '@itwin/itwinui-icons-react/cjs/icons/ChevronRight';
+import { ButtonGroup } from '../ButtonGroup';
+import { Button } from '../Buttons/Button';
+import { useOverflow } from '../utils/hooks/useOverflow';
+import { ProgressRadial } from '../ProgressIndicators';
+import { TableProps } from './Table';
+import { DropdownButton } from '../Buttons';
+import { MenuItem } from '../Menu';
+import { CommonProps, getBoundedValue, useTheme } from '../utils';
+
+const defaultLocalization = {
+  rowsPerPageLabel: (size: number) => `${size} per page`,
+  currentRowsInfoLabel: (
+    startIndex: number,
+    endIndex: number,
+    totalRows: number,
+    isLoading: boolean,
+  ) =>
+    isLoading
+      ? `${startIndex}-${endIndex}…`
+      : `${startIndex}-${endIndex} of ${totalRows}`,
+};
+
+export type TablePaginatorProps = {
+  /**
+   * The zero-based index of the current page.
+   */
+  currentPage: number;
+  /**
+   * Total number of rows.
+   */
+  totalRowsCount: number;
+  /**
+   * Number of rows per page.
+   */
+  pageSize: number;
+  /**
+   * Callback when page is changed.
+   */
+  onPageChange: (page: number) => void;
+  /**
+   * Callback when page size is changed.
+   */
+  onPageSizeChange?: (size: number) => void;
+  /**
+   * Modify the density of the pagination (adjusts the height).
+   * @default 'default'
+   */
+  density?: TableProps['density'];
+  /**
+   * Flag whether data is still loading and total rows count is not known.
+   * @default false
+   */
+  isLoading?: boolean;
+  /**
+   * Control whether focusing tabs (using arrow keys) should automatically select them.
+   * Use 'manual' if tab panel content is not preloaded.
+   * @default 'auto'
+   */
+  focusActivationMode?: 'auto' | 'manual';
+  /**
+   * Array of possible page size options. When provided then shows current rows info and page size selection.
+   */
+  rowsPerPage?: number[];
+  /**
+   * Object of functions to localize pagination labels.
+   */
+  localization?: {
```

Comment:
Not sure about the naming of this and functions inside.

Reply:
Check this out: https://material.angular.io/components/paginator/api

Reply:
Ok, I just went trough our project and found out that we are using `localization` and `translated...` for our props. We really need to decide which one to use.